client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/ReindexCacheTest.java

```diff
@@ -0,0 +1,127 @@
+import org.infinispan.query.MassIndexer;
+import org.infinispan.query.backend.QueryInterceptor;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.remote.ProtobufMetadataManager;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for the MassIndexer when using storing protobuf
+ */
+@Test(testName = "client.hotrod.query.ReindexCacheTest", groups = "functional")
+public class ReindexCacheTest extends SingleHotRodServerTest {
+
+   private static final String USER_CACHE = "users";
+
+   private static final int NUM_ENTRIES = 50;
+
+   private StorageType storageType;
+
+   @Factory
+   public Object[] factory() {
+      return new Object[]{
+            new ReindexCacheTest().storageType(StorageType.OBJECT),
+            new ReindexCacheTest().storageType(StorageType.OFF_HEAP)
+      };
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      cacheManager = createServerModeCacheManager(hotRodCacheConfiguration());
+      cacheManager.defineConfiguration(USER_CACHE, hotRodCacheConfiguration(buildIndexedConfig()).build());
+      return cacheManager;
+   }
+
+   @Override
+   protected String parameters() {
+      return "storageType-" + storageType;
+   }
+
+   private ReindexCacheTest storageType(StorageType storageType) {
+      this.storageType = storageType;
+      return this;
+   }
+
+   public ConfigurationBuilder buildIndexedConfig() {
+      ConfigurationBuilder builder = hotRodCacheConfiguration(new ConfigurationBuilder());
+      builder.memory().storageType(storageType).indexing().index(ALL)
+            .addProperty("default.directory_provider", "local-heap");
+      return builder;
+   }
+
+   @BeforeClass(alwaysRun = true)
+   protected void registerSerCtx() throws Exception {
+      ProtobufMetadataManager protobufMetadataManager = cacheManager.getGlobalComponentRegistry().getComponent(ProtobufMetadataManager.class);
+      String proto = Util.getResourceAsString("/sample_bank_account/bank.proto", getClass().getClassLoader());
+      protobufMetadataManager.registerProtofile("bank.proto", proto);
+      MarshallerRegistration.registerMarshallers(remoteCacheManager);
+   }
+
+   @Test
+   public void testMassIndexing() {
+      RemoteCache<Integer, UserPB> userCache = remoteCacheManager.getCache(USER_CACHE);
+      getUsersPB().forEach(u -> userCache.put(u.getId(), u));
+
+      assertEquals(query(userCache), NUM_ENTRIES);
+
+      wipeIndexes();
+      assertIndexEmpty();
+      reindex();
+
+      assertEquals(query(userCache), NUM_ENTRIES);
+   }
+
+   private void wipeIndexes() {
+      AdvancedCache<?, ?> advancedCache = cacheManager.getCache(USER_CACHE).getAdvancedCache();
+      QueryInterceptor qi = advancedCache.getComponentRegistry().getComponent(QueryInterceptor.class);
+      qi.purgeAllIndexes();
+   }
+
+   private void assertIndexEmpty() {
```

Comment:
is this method necessary? instead of just calling assertEquals(query(userCache), 0);